core-blocks/latest-comments/index.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Server-side rendering of the `core/latest-comments` block.
+ *
+ * @package gutenberg
+ */
+
+// Used to get `_draft_or_post_title`, which prevents empty-title posts from
+// appearing with no title.
+require_once( ABSPATH . 'wp-admin/includes/template.php' );
+
+$default_comments_to_show = 5;
+
+/**
+ * Renders the `core/latest-comments` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the post content with latest comments added.
+ */
+function gutenberg_render_block_core_latest_comments( $attributes = array() ) {
+	// Basic attribute validation.
+	if (
+		! isset( $attributes['align'] ) ||
+		! in_array( $attributes['align'], array( 'center', 'left', 'right', 'wide', 'full' ), true )
+	) {
+		$attributes['align'] = null;
+	}
+	if (
+		! is_numeric( $attributes['commentsToShow'] ) ||
+		$attributes['commentsToShow'] < 0 ||
+		$attributes['commentsToShow'] > 100
+	) {
+		$attributes['commentsToShow'] = $default_comments_to_show;
+	}
+
+	// This filter is documented in wp-includes/widgets/class-wp-widget-recent-comments.php.
+	$comments = get_comments( apply_filters( 'widget_comments_args', array(
+		'number'      => $attributes['commentsToShow'],
+		'status'      => 'approve',
+		'post_status' => 'publish',
+	) ) );
+
+	$list_items_markup = '';
+	if ( ! empty( $comments ) ) {
+		// Prime the cache for associated posts. This is copied from \WP_Widget_Recent_Comments::widget().
+		$post_ids = array_unique( wp_list_pluck( $comments, 'comment_post_ID' ) );
+		_prime_post_caches( $post_ids, strpos( get_option( 'permalink_structure' ), '%category%' ), false );
+
+		foreach ( $comments as $comment ) {
+			$list_items_markup .= '<li class="wp-block-latest-comments__comment">';
+			if ( $attributes['displayAvatar'] ) {
+				$avatar = get_avatar( $comment, 48, '', '', array(
+					'class' => 'wp-block-latest-comments__comment-avatar',
+				) );
+				if ( $avatar ) {
+					$list_items_markup .= $avatar;
+				}
+			}
+
+			$list_items_markup .= '<article>';
+			$list_items_markup .= '<footer class="wp-block-latest-comments__comment-meta">';
+			$author_url         = get_comment_author_url( $comment );
+			if ( empty( $author_url ) && ! empty( $comment->user_id ) ) {
+				$author_url = get_author_posts_url( $comment->user_id );
+			}
+
+			$author_markup = '';
+			if ( $author_url ) {
+				$author_markup .= '<a class="wp-block-latest-comments__comment-author" href="' . esc_url( $author_url ) . '">' . get_comment_author( $comment ) . '</a>';
+			} else {
+				$author_markup .= '<span class="wp-block-latest-comments__comment-author">' . get_comment_author( $comment ) . '</span>';
+			}
+
+			$post_title = '<a class="wp-block-latest-comments__comment-link" href="' . esc_url( get_comment_link( $comment ) ) . '">' . esc_html( _draft_or_post_title( $comment->comment_post_ID ) ) . '</a>';
```

Comment:
`_draft_or_post_title` already calls `esc_html`, though I can accept if you'd rather also escape here for reviewers peace of mind.

Reply:
I think I originally added the `esc_html` (or it was here already?) because we weren't using `_draft_or_post_title`. I suppose now that it's being used we can omit the extra escape.

In general I like seeing it but in this case I'll leave a comment explaining that `_draft_or_post_title` does the escaping. No need to run it twice.